dll/win32/msgina/shutdown.c

```diff
@@ -848,140 +872,141 @@ ShutdownOnInit(
-
-    hwndList = GetDlgItem(hDlg, IDC_SHUTDOWN_ACTION);
-
-    /* Clear the content before it's used */
-    SendMessageW(hwndList, CB_RESETCONTENT, 0, 0);
+    if (pContext->bFriendlyUI)
+    {
+        /* Create font for the IDC_TURN_OFF_STATIC static control */
+        hdc = GetDC(hDlg);
+        lfHeight = -MulDiv(FONT_POINT_SIZE, GetDeviceCaps(hdc, LOGPIXELSY), 72);
+        ReleaseDC(hDlg, hdc);
+        pContext->hfFont = CreateFontW(lfHeight, 0, 0, 0, FW_MEDIUM, FALSE, 0, 0, 0, 0, 0, 0, 0, L"MS Shell Dlg");
+        SendDlgItemMessageW(hDlg, IDC_TURN_OFF_STATIC, WM_SETFONT, (WPARAM)pContext->hfFont, TRUE);
+
+        /* Create a brush for static controls for fancy shut down dialog */
+        pContext->hBrush = CreateSolidBrush(DARK_GREY_COLOR);
+
+        /* Gather image strip */
+        pContext->hImageStrip = LoadBitmapW(pgContext->hDllInstance, MAKEINTRESOURCEW(IDB_IMAGE_STRIP));
+        
+        /* Set the boolean flags to false */
+        pContext->bIsSleepButtonReplaced = FALSE;
+        pContext->bTimer = FALSE;
+
+        for (int i = 0; i < NUMBER_OF_BUTTONS; i++)
+        {
+            pContext->bIsButtonHot[i] = FALSE;
+        }
 
-    /* Set the boolean flags to false */
-    pContext->bIsSleepButtonReplaced = FALSE;
+       EnableWindow(GetDlgItem(hDlg, IDC_BUTTON_HIBERNATE), FALSE);
+       EnableWindow(GetDlgItem(hDlg, IDC_BUTTON_SLEEP), IsPwrSuspendAllowed());
+
+        /* Create tool tips for the buttons of fancy log off dialog */
+        CreateToolTipForButtons(IDC_BUTTON_HIBERNATE,
+                                IDS_SHUTDOWN_HIBERNATE_DESC,
+                                hDlg, IDS_SHUTDOWN_HIBERNATE,
+                                pContext->pgContext->hDllInstance);
+        CreateToolTipForButtons(IDC_BUTTON_SHUTDOWN,
+                                IDS_SHUTDOWN_SHUTDOWN_DESC,
+                                hDlg, IDS_SHUTDOWN_SHUTDOWN,
+                            pContext->pgContext->hDllInstance);
+        CreateToolTipForButtons(IDC_BUTTON_REBOOT,
+                                IDS_SHUTDOWN_RESTART_DESC,
+                                hDlg, IDS_SHUTDOWN_RESTART,
+                                pContext->pgContext->hDllInstance);
+        CreateToolTipForButtons(IDC_BUTTON_SLEEP,
+                                IDS_SHUTDOWN_SLEEP_DESC,
+                                hDlg, IDS_SHUTDOWN_SLEEP,
+                                pContext->pgContext->hDllInstance);
+
+        /* Gather old button func */
+        pContext->OldButtonProc = (WNDPROC)GetWindowLongPtrW(GetDlgItem(hDlg, IDC_BUTTON_HIBERNATE), GWLP_WNDPROC);
+
+        /* Make buttons to remember pContext and subclass the buttons */
+        for (int i = 0; i < NUMBER_OF_BUTTONS; i++)
+        {
+            SetWindowLongPtrW(GetDlgItem(hDlg, IDC_BUTTON_HIBERNATE + i), GWLP_USERDATA, (LONG_PTR)pContext);
+            SetWindowLongPtrW(GetDlgItem(hDlg, IDC_BUTTON_HIBERNATE + i), GWLP_WNDPROC, (LONG_PTR)OwnerDrawButtonSubclass);
```

Comment:
Why not use `SetWindowSubclass` instead?

Reply:
MSGINA shouldn't depend on components like Comctl32 to work, which `SetWindowSubclass` depends upon.